redisson/src/main/java/org/redisson/spring/session/RedissonSessionRepository.java

```diff
@@ -226,6 +226,8 @@ public String changeSessionId() {
             batchNew.getMap(keyPrefix + id, map.getCodec()).putAllAsync(oldState);
             if (remainTTL > 0) {
                 batchNew.getBucket(getExpiredKey(id)).setAsync("", remainTTL, TimeUnit.MILLISECONDS);
+                // Add 60 seconds to the map expiration, to match updateExpiration()
+                batchNew.getMap(keyPrefix + id, map.getCodec()).expireAsync(remainTTL + 60 * 1000L, TimeUnit.MILLISECONDS);
```

Comment:
We’re almost duplicating `updateExpiration()` here, with two key differences:
- we’re copying the existing TTls, rather than setting to the maximum
- we’re using a batch, so it will be faster

However, I think we’d be OK to just call `updateExpiration()`.